synapse/rest/admin/rooms.py

```diff
@@ -306,10 +308,13 @@ async def on_GET(
             raise NotFoundError("Room not found")
 
         members = await self.store.get_users_in_room(room_id)
-        ret["joined_local_devices"] = await self.store.count_devices_by_users(members)
-        ret["forgotten"] = await self.store.is_locally_forgotten_room(room_id)
+        result = attr.asdict(ret)
+        result["joined_local_devices"] = await self.store.count_devices_by_users(
+            members
+        )
+        result["forgotten"] = await self.store.is_locally_forgotten_room(room_id)
```

Comment:
Is this more classic synapse logic in the rest layer?

Reply:
I think for the admin API we tend to do that a bit more?